abd-clam/src/core/dataset/vec2d.rs

```diff
@@ -206,4 +368,64 @@ mod tests {
             Some([12].as_slice())
         );
     }
+
+    #[test]
+    fn test_save_load_deterministic() {
+        let data = vec![vec![1, 2, 3, 4, 5], vec![6, 7, 8, 9, 10]];
+        let data: Vec<&[u32]> = data.iter().map(Vec::as_slice).collect();
+
+        let mut dataset = VecDataset::new("test".to_string(), data, euclidean_sq::<u32, u32>, false);
+        let indices = dataset.indices().iter().map(|x| *x).rev().collect::<Vec<usize>>();
+        dataset.set_reordered_indices(&indices);
+        dataset.save(&std::env::temp_dir().join("datset.save")).unwrap();
```

Comment:
Add and use the `tempdir` [crate](https://docs.rs/tempdir/latest/tempdir/) as a dev dependency.